mevshare/sim_result_backend.go

```diff
@@ -56,38 +56,41 @@ func (s *SimulationResultBackend) SimulatedBundle(ctx context.Context,
 	}
 	logger := s.log.With(zap.String("bundle", hash.Hex()))
 
-	// failed bundle does not go to the builder
-	err := s.store.InsertBundleForStats(ctx, bundle, sim)
-	if err != nil {
-		if errors.Is(err, ErrKnownBundle) {
-			logger.Debug("Bundle already known", zap.Error(err))
-			return nil
+	go func(ctx context.Context) {
```

Comment:
nit: think it's worth adding waitgroup and waiting for this to complete before we return